### ISSUE TITLE
Build failure on macOS

### ISSUE THREAD

**Issue by jserv** [RinHizakura/mini-gdbstub#1]
While attempting to build latest rv32emu + mini-gdbstub on macOS Monterey, I got the following messages:
```
In file included from ./lib/gdbstub.c:7:
include/gdb_signal.h:1:9: error: 'GDB_SIGNAL' is used as a header guard here, followed by #define of a different macro [-Werror,-Wheader-guard]
#ifndef GDB_SIGNAL
        ^~~~~~~~~~
include/gdb_signal.h:2:9: note: 'GDG_SIGNAL' is defined here; did you mean 'GDB_SIGNAL'?
#define GDG_SIGNAL
        ^~~~~~~~~~
        GDB_SIGNAL
1 error generated.
```

Replacing the use of `GDB_SIGNAL` with `#pragma once` would pass.

**Comment by RinHizakura** [comment]
The reason looks like a typo on `GD*G*_SIGNAL` which should be ` GD*B*_SIGNAL`. Could you take a try on the fix in [db20f16](https://github.com/RinHizakura/mini-gdbstub/commit/db20f16e5f7782ce7c479ff8b62a4164b6d9029a)?

**Comment by jserv** [comment]
> The reason looks like a typo on `GD*G*_SIGNAL` which should be ` GD*B*_SIGNAL`. Could you take a try on the fix in [db20f16](https://github.com/RinHizakura/mini-gdbstub/commit/db20f16e5f7782ce7c479ff8b62a4164b6d9029a)?

Verified. It works now.
